docs/v2_migration.md

```diff
@@ -0,0 +1,267 @@
+
+## Python **3** only
+
+First and main change is the drop of python 2 support. To be honest we first started this lib as a python 3 only and then switch back to support python 2. Now we are in 2020 and python 2 is [officially dead](https://pythonclock.org) we decided to remove python 2 support and to continue with only python 3.
+
+If you need help moving from python 2 to 3 checkout the official [doc](https://docs.python.org/3/howto/pyporting.html).
+
+### Type hints
+
+By switching to python 3 we also embrace new code style with the adoption of type hints:
+
+    Python 3.6+ has support for optional "type hints".
+    These "type hints" are a new syntax (since Python 3.6+) that allow declaring the type of a variable.
+    By declaring types for your variables, editors and tools can give you better support.
+from:  https://fastapi.tiangolo.com/python-types/
+
+Other docs:
+- https://kishstats.com/python/2019/01/07/python-type-hinting.html
+- https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html
+
+Example:
+```python
+def get_vrt_transform(
+    src_dst: Union[DatasetReader, DatasetWriter, WarpedVRT],
+    bounds: Tuple[float, float, float, float],
+    dst_crs: CRS = constants.WEB_MERCATOR_CRS,
+) -> Tuple[Affine, int, int]:
+```
+
+## Rasterio >= **1.1.3**
+
+Recent changes in rasterio makes masking more reliable.
+
+## New **rio_tiler.io** submodules
+
+Mostly to gain in code clarity, we moved the mission specific submodules (e.g. rio_tiler.landsat8) to an `rio_tiler.io` submodule. `rio_tiler.main` has also be renamed `rio_tiler.io.cogeo`.
+
+```python
+# v1
+from rio_tiler.main import tile as cogTiler
+tile, mask = cogTiler('my_tif.tif', 691559, 956905, 21, tilesize=256)
+
+# v2
+from rio_tiler.io.cogeo import tile as cogTiler
+tile, mask = cogTiler('my_tif.tif', 691559, 956905, 21, tilesize=256)
+```
+
+```python
+# v1
+from rio_tiler import landsat8
+landsat8.bounds('LC08_L1TP_016037_20170813_20170814_01_RT')
+
+# v2
+from rio_tiler.io import landsat8
+landsat8.bounds('LC08_L1TP_016037_20170813_20170814_01_RT')
+```
+
+## **rio_tiler.reader**
+
+Internal tile/data reading functions have been refactored and moved to a new `rio_tiler.reader` submodule.
+
+### tile
+
+In *rio_tiler==1* most of the magic was happening in [`rio_tiler.utils._tile_read`](https://github.com/cogeotiff/rio-tiler/blob/master/rio_tiler/utils.py#L337-L349). In *rio_tiler==2* this function is now splited in 2 `rio_tiler.reader.part` and `rio_tiler_reader._read` to reduce code reutilisation and make code more robust. The `part` function now takes `height` and `width` instead of a unique `tilesize` to specify the output array size. 
+
+To ease the transition we added `rio_tiler.reader.tile` function.
+
+Note: The new `rio_tiler.reader.part` function enables to perform non-squared data cropping of different.
```

Comment:
Funny sentence - Not sure what the intent is?

Reply:
me neither 🤦‍♂

Reply:
Will add 👇 

Note: The new `rio_tiler.reader.part` function enables to perform non-squared data cropping by passing output **width** and **height** (instead of just `tilesize`).